src/api/Feed.js

```diff
@@ -95,6 +103,75 @@ const getItemWithPendingReply = <T: { replies?: Array<Comment> }>(item: T, reply
     return { replies: [...replies, reply], ...rest };
 };
 
+const parseFileActivitiesResponseForFeed = (response?: { entries: FileActivity[] }) => {
+    if (!response || !response.entries || !response.entries.length) {
+        return [];
+    }
+
+    const data = response.entries;
+    const parsedData = [];
+
+    data.map(item => {
+        if (!item.source) {
+            return item;
+        }
+
+        const source = { ...item.source };
+        // $FlowFixMe
+        const commentItem: Comment = source[FILE_ACTIVITY_TYPE_COMMENT];
+        // $FlowFixMe
+        const taskItem: Task = source[FILE_ACTIVITY_TYPE_TASK];
+
+        switch (item.type) {
+            case FILE_ACTIVITY_TYPE_TASK:
+                // UAA follows a lowercased enum naming convention, convert to uppercase to align with task api
+                if (!!taskItem.assigned_to && !!taskItem.assigned_to.entries) {
```

Comment:
Does BUIE support optional chaining? Is the `!!` needed?

Reply:
I believe it does based on other usages in BUIE, linter removes the `!!` after optional chaining too